routing_model/src/actions.rs

```diff
@@ -79,20 +81,25 @@ impl InnerAction {
     }
 
     pub fn with_section_members(mut self, section: SectionInfo, nodes: &[Node]) -> Self {
-        self.section_members.insert(section, nodes.to_vec());
+        assert!(self
```

Comment:
I'm not super happy to see that whole statement in an assert.
I'd rather have:
```
let inserted = self.section_members.insert(section, nodes.to_vec());
assert!(inserted.is_none());
```

Or if it is just to shut up the warning:
```let _ = ...```

I'm not sure this assert bring a sensible restriction to this function.

Reply:
I'm still pretty new to the code in here, but it didn't seem reasonable to me to allow that insertion to replace an existing element.  If that _is_ expected, I'd prefer some comment as to the rationale (I can add such a comment if you think that is to be expected).

I think the clippy warning is useful, in that it's at least making us consider whether we really mean to silently drop existing entries.

I'll update to create a temporary `inserted` as suggested for now.